app/src/lib/stores/app-store.ts

```diff
@@ -4352,7 +4352,18 @@ export class AppStore extends TypedBaseStore<IAppState> {
     files: ReadonlyArray<WorkingDirectoryFileChange>
   ) {
     const gitStore = this.gitStoreCache.get(repository)
-    await gitStore.discardChanges(files)
+
+    try {
+      await gitStore.discardChanges(files)
+    } catch (error) {
+      log.error('Failed discarding changes', error)
+
+      this.emitError(
+        new Error(
+          `Failed to discard changes to ${TrashNameLabel}.\n\nA common reason for this is that the directory or one of its files is open in another program or the ${TrashNameLabel} is set to delete items immediately.`
```

Comment:
Notes.. I am not sure if the "A common reason for this is that the directory or one of its files is open in another program" applies here as I think that is a problem more so for directory level deletions..